src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/ContentPickerFieldDisplayDriver.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using OrchardCore.ContentFields.Settings;
+using OrchardCore.ContentFields.ViewModels;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.ContentManagement.Display.Models;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using YesSql;
+using YesSql.Services;
+
+namespace OrchardCore.ContentFields.Fields
+{
+    public class ContentPickerFieldDisplayDriver : ContentFieldDisplayDriver<ContentPickerField>
+    {
+        private readonly IContentManager _contentManager;
+        private readonly ISession _session;
+
+        public ContentPickerFieldDisplayDriver(
+            IContentManager contentManager,
+            ISession session,
+            IStringLocalizer<ContentPickerFieldDisplayDriver> localizer)
+        {
+            _contentManager = contentManager;
+            _session = session;
+            T = localizer;
+        }
+
+        public IStringLocalizer T { get; set; }
+
+        public override IDisplayResult Display(ContentPickerField field, BuildFieldDisplayContext context)
+        {
+            return Initialize<DisplayContentPickerFieldViewModel>("ContentPickerField", async model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+                model.Updater = context.Updater;
+                model.SelectedContentItemIds = string.Join(",", field.ContentItemIds);
+                model.ContentItems = await _session.Query<ContentItem, ContentItemIndex>()
```

Comment:
Please use `IContentManager.GetAsync` I just added one that accepts a list of ids. This is necessary to trigger `Load` events on the content items.